app/src/components/circular-progress.tsx

```diff
@@ -0,0 +1,105 @@
+import { ReactElement } from 'react';
+
+export const cleanPercentage = (percentage: number | string): number => {
+  const isNegativeOrNaN = !Number.isFinite(+percentage) || +percentage < 0; // we can set non-numbers to 0 here
+  const isTooHigh = +percentage > 100;
+  return isNegativeOrNaN ? 0 : isTooHigh ? 100 : +percentage;
+};
+
+export const ProgressCircle = ({
+  colour,
+  percentage,
+}: {
+  colour: string;
+  percentage: number | string;
+}): ReactElement => {
+  // An SVG circle with stroke-width renders two concentric circles (or a doughnut like image)
+  // with its internal radius as r - ½ the stroke width and its external radius as r + ½ stroke-width.\
+  // For a circle with d=400, r should be (r - strokeWidth / 2)
+  const r = 197.5; // (r * 2) + strokeWidth needs to equal the size of the intended circle - this this case 410
+  const circ = 2 * Math.PI * r;
+  const strokePercentage = ((100 - +percentage) * circ) / 100; // where stroke will start, e.g. from 15% to 100%.
+  return (
+    <circle
+      data-testid={`progress-circle-${colour}-test`}
+      r={r}
+      cx={-5}
+      // cy={-105} // Can remove this and inclde in transform translate attribute
+      fill="transparent"
+      stroke={strokePercentage !== circ ? colour : ''} // remove colour as 0% sets full circumference
```

Comment:
Lets remove too much logic happening inside the property and simplify it by adding to a variable first then passing the variable to the prop.

```ts
const stroke = strokePercentage !== circ ? colour : ''
```